packages/fields/src/components/create-template-part-modal/index.tsx

```diff
@@ -79,11 +130,14 @@ export function CreateTemplatePartModalContents( {
 	const [ isSubmitting, setIsSubmitting ] = useState( false );
 	const instanceId = useInstanceId( CreateTemplatePartModal );
 
-	const templatePartAreas = useSelect(
-		( select ) =>
-			select( editorStore ).__experimentalGetDefaultTemplatePartAreas(),
+	const settings = useSelect(
+		// @ts-ignore
+		( select ) => select( blockEditorStore ).getSettings(),
 		[]
 	);
+
+	const defaultTemplatePartAreas = getDefaultTemplatePartAreas( settings );
```

Comment:
Personally, I think this is the wrong solution to retrieving the default template areas. There might not be a rendered block-editor when using the fields, so I don't think a dependency to "block-editor" is the right approach.

The main issue here is that `defaultTemplateTypes` and `defaultTemplatePartAreas ` are passed from the server to the client through the editor settings initialization. I think this is the wrong approach. The right approach would be to move these to `getEntityRecord( 'root', '__unstableBase' )` endpoint and add them in the server to this endpoint. I believe we did things like that in the past for things like `site_icon_url` ...  Do you think this is something we can try in its own PR. I'm a bit concerned that the current field won't work as expected in all contexts.

Reply:
> Do you think this is something we can try in its own PR. I'm a bit concerned that the current field won't work as expected in all contexts.

I'm happy to work on it with a dedicated PR! 💪

Reply:
I created #66459. Let me know if you agree with the approach that I took!